app/login/page.tsx

```diff
@@ -0,0 +1,59 @@
+import "dotenv/config"
+import bcrypt from "bcrypt"
+import { redirect } from "next/navigation"
+import jwt from "jsonwebtoken"
+import { cookies } from "next/headers"
+
+import AuthForm from "../components/AuthForm"
+import { client } from "../utils/mongodb/newClient"
+
+export default function Login() {
+  async function login(formData: FormData) {
+    "use server"
+    const data = {
+      username: formData.get("username"),
+      password: formData.get("password"),
+    }
+
+    const username = data.username
+    const password = data.password
+
+    if (typeof username !== "string" || typeof password !== "string") {
```

Comment:
don't you want some error handling here to alert the user? and when you're checking for user and password below too?